src/codegen/compile.ml

```diff
@@ -2038,72 +1971,55 @@ module Tagged = struct
     | BigInt -> 23l
     | Concat -> 25l
     | Region -> 27l
-    | Null -> 29l
-    | OneWordFiller -> 31l
-    | FreeSpace -> 33l
+    | OneWordFiller -> 29l
+    | FreeSpace -> 31l
     | ArraySliceMinimum -> 34l
     (* Next two tags won't be seen by the GC, so no need to set the lowest bit
        for `CoercionFailure` and `StableSeen` *)
     | CoercionFailure -> 0xfffffffel
     | StableSeen -> 0xffffffffl
 
-  (* Declare `env` for lazy computation of the header size when the compile environment with compile flags are defined *)
-  let header_size env =
-    if !Flags.gc_strategy = Flags.Incremental then 2l else 1l
+  (*
+     The null pointer is the sentinel `0xffff_fffbl` (skewed representation).
+    
+     This serves for efficient null tests by using direct pointer comparison.
+     The null pointer must not be dereferenced.
+     Null tests are possible without resolving the forwarding pointer of a non-null comparand.
+  *)
+
+  let null_vanilla_pointer = 0xffff_fffbl (* skewed, pointing to last unallocated Wasm page *)
```

Comment:
Is this actually pointing to the last word of the last unallocated page?

Reply:
Yes, I believe so, skewed pointer to last word in 32-bit space. Do you think a different value should be chosen, e.g. the middle of the last page to protect cases where unchecked small-range pointer arithmetics are applied to `null`?
(In 64-bit mode, I even kept `0xffff_ffff_ffff_ffffbL`, pointing in the middle of the last word.)

Reply:
Not the comment was just a bit vague. It's fine.